### DISCUSSION TITLE
Connecting to DAX Using Only PHP (Without AWS SDK)

### DISCUSSION THREAD

**Discussion by raphaelcangucu** [aws/aws-sdk-php#2950] [Q&A]
Hello everyone,

I hope you are all doing well.

I am currently working on a project where we extensively use DynamoDB and are experiencing performance lags. We attempted to leverage DynamoDB Accelerator (DAX) to mitigate this issue but encountered a significant roadblock: the AWS SDK for PHP does not currently support querying a DAX cluster.

We have been unable to find any example of a simple PHP function to query DAX without using the SDK. I wanted to reach out to the community to ask if anyone has managed to connect to a DAX cluster using only PHP, without relying on the AWS SDK.

If you have any examples, scripts, or libraries that you could share, it would be incredibly helpful. For reference, here is the GitHub issue discussing this problem: [GitHub Issue #2758](https://github.com/aws/aws-sdk-php/issues/2758).

Thank you in advance for your assistance!

Best regards,

Raphael

**Comment by yenfryherrerafeliz** [comment]
Hi @raphaelcangucu, thanks for reaching out. I see the value of having  DAX support in the PHP SDK, however, the DAX SDKs are handled by a different internal team and therefore at this time I do not have an answer for you regarding if having DAX support on PHP SDK will be possible in the near future. Nonetheless, I have reached out to that internal team and I am waiting for them to answer. Once I get some updates from them I will make to provide those to you here on this discussion.

Thanks!

**Reply by raphaelcangucu** [reply]
Hi @yenfryherrerafeliz , thanks for your prompt reply.

We also reached the AWS Tech Support to see if they can handle this too.

Also I am trying to write using the NodeJs API:

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DAX.concepts.html

<img width="838" alt="image" src="https://github.com/aws/aws-sdk-php/assets/5266252/9afa2772-4cc2-408c-b140-462d6e5a8b03">

And I am getting slower time than using the orginal DynamoDb Connection:

```bash
~/TryDax/nodejs$ node 07-game-match-test.js daxs://daxtest.1dr0hr.dax-clusters.us-east-1.amazonaws.com
Put succeeded: {}
        Total time:  291 ms 
~/TryDax/nodejs$ node 07-game-match-test.js 
Put succeeded: {}
        Total time:  48 ms 
forge@calm-sunrise:~/pyDaxClient/
```

It's five times slower than the Dynamo... Do you know why?
